File: 0019-remove-nth-node-from-end-of-list/0019-remove-nth-node-from-end-of-list.cpp
class Solution {
public:
    ListNode* removeNthFromEnd(ListNode* head, int n) {
    ListNode*temp=head;
    ListNode*dummy=new ListNode(0);
    dummy->next=head;
    ListNode*curr=dummy;
    while(temp!=NULL&&n!=0){
        temp=temp->next;
        n--;
    }       
    while(temp!=NULL){
        temp=temp->next;
        curr=curr->next;
    } 
    ListNode*front=curr->next;
    curr->next=front->next;
    front->next=NULL;
    return dummy->next;
    }
};

File: 0019-remove-nth-node-from-end-of-list/0019-remove-nth-node-from-end-of-list.py
# Definition for singly-linked list.
# class ListNode(object):
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution(object):
    def removeNthFromEnd(self, head, n):
        dummy = ListNode(0)
        dummy.next = head
        prev=dummy
        curr=dummy
        while n:
            curr=curr.next
            n=n-1
        while curr.next is not None:
            prev=prev.next
            curr=curr.next
        prev.next=prev.next.next
        return dummy.next